stdlib/public/core/Collection.swift

```diff
@@ -398,6 +392,52 @@ extension Collection {
   }
 }
 
+/// Supply optimized defaults for `Collection` models that use some model 
+/// of `Strideable` as their `Index`.
+extension Collection where Index : Strideable {
+  @warn_unused_result
+  public func next(i: Index) -> Index {
+    _failEarlyRangeCheck(i, bounds:startIndex..<endIndex)
+
+    return i.advanced(by: 1)
+  }
+
+  /*
+  @warn_unused_result
+  public func advance(i: Index, by n: IndexDistance) -> Index {
+    _precondition(n >= 0,
+      "Only BidirectionalCollections can be advanced by a negative amount")
+    // FIXME: swift-3-indexing-model: range check i
+
+    // FIXME: swift-3-indexing-model - error: cannot invoke 'advanced' with an argument list of type '(by: Self.IndexDistance)'
+    return i.advanced(by: n)
```

Comment:
Not sure how to deal with the compile errors I get here (and below). I think I understand why just not the best way to deal with it.